### ISSUE TITLE
Attempt to invoke virtual method 'offscreenAnnotationViewContainer()' on a null object reference RCTMGLPointAnnotation.removeFromMap

### ISSUE THREAD

**Issue by flo5324** [react-native-mapbox-gl/maps#909]
**Describe the bug**
This crash happens in my release build. It seems to be a silent crash that doesn't affect react native. But this bug occurs quite a lot.
The app consist of a map displaying PointAnnotation for locations and routes.
The locations are generated from a map() on an array. Markers can be removed or added with filters.

This bug seems to happen more ofter on Samsung and Huawei devices but not on a particular android version.

As I understand, the null comes from the function _offscreenAnnotationViewContainer()_ which is used to "cache?" PointAnnotations. So I think it might comes from a problem with the lifecycle of this view and its parent.

Maybe i'm just doing something wrong ?

**To Reproduce**
I was not able to reproduce it on my Pixel 4 in dev mode.

**Versions (please complete the following information):**
 - Platform: Android
 - Device: Samsung, Huawei, etc
 - Emulator/ Simulator: no
 - OS: Android 5 => 10
 - react-native-mapbox-gl Version 8.0.0
 - React Native Version 0.61.5

**Additional context**
For what it's worth, react-native-screens is not enabled on this app, i'll try with it enabled.
I'm sorry I can't provide an example as my application is quite massive and also I can't reproduce it on my device.

Full stack trace : 
```
Fatal Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'android.view.ViewGroup com.mapbox.rctmgl.components.mapview.RCTMGLMapView.offscreenAnnotationViewContainer()' on a null object reference
       at com.mapbox.rctmgl.components.annotation.RCTMGLPointAnnotation.removeFromMap(RCTMGLPointAnnotation.java:120)
       at com.mapbox.rctmgl.components.mapview.RCTMGLMapView.removeFeature(RCTMGLMapView.java:268)
       at com.mapbox.rctmgl.components.mapview.RCTMGLMapViewManager.removeViewAt(RCTMGLMapViewManager.java:92)
       at com.mapbox.rctmgl.components.mapview.RCTMGLMapViewManager.removeViewAt(RCTMGLMapViewManager.java:39)
       at com.facebook.react.uimanager.NativeViewHierarchyManager.manageChildren(NativeViewHierarchyManager.java:457)
       at com.facebook.react.uimanager.UIViewOperationQueue$ManageChildrenOperation.execute(UIViewOperationQueue.java:205)
       at com.facebook.react.uimanager.UIViewOperationQueue$1.run(UIViewOperationQueue.java:779)
       at com.facebook.react.uimanager.UIViewOperationQueue.flushPendingBatches(UIViewOperationQueue.java:888)
       at com.facebook.react.uimanager.UIViewOperationQueue.access$2200(UIViewOperationQueue.java:42)
       at com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.doFrameGuarded(UIViewOperationQueue.java:948)
       at com.facebook.react.uimanager.GuardedFrameCallback.doFrame(GuardedFrameCallback.java:28)
       at com.facebook.react.modules.core.ReactChoreographer$ReactChoreographerDispatcher.doFrame(ReactChoreographer.java:174)
       at com.facebook.react.modules.core.ChoreographerCompat$FrameCallback$1.doFrame(ChoreographerCompat.java:84)
       at android.view.Choreographer$CallbackRecord.run(Choreographer.java:909)
       at android.view.Choreographer.doCallbacks(Choreographer.java:723)
       at android.view.Choreographer.doFrame(Choreographer.java:655)
       at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:897)
       at android.os.Handler.handleCallback(Handler.java:790)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loop(Looper.java:164)
       at android.app.ActivityThread.main(ActivityThread.java:7000)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:441)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1408)
```

**Comment by flo5324** [comment]
I just discovered that this might be fixed with #908 
I'll switch to this version but this looks promising

**Comment by mfazekas** [comment]
Should be fixed by  #908
